scalafix-rules/src/main/scala/scalafix/internal/rule/CompilerTypePrinter.scala

```diff
@@ -45,13 +45,11 @@ class CompilerTypePrinter(g: ScalafixGlobal, config: ExplicitResultTypesConfig)(
       space: String
   ): Option[v1.Patch] = {
     val gpos = unit.position(pos.start)
-    val tpe = GlobalProxy.typedTreeAt(g, gpos)
-    val inverseSemanticdbSymbol =
-      if (sym.isLocal) tpe.symbol
-      else
-        g.inverseSemanticdbSymbols(sym.value)
-          .find(s => g.semanticdbSymbol(s) == sym.value)
-          .getOrElse(g.NoSymbol)
+    GlobalProxy.typedTreeAt(g, gpos)
```

Comment:
@olafurpg is this called just for its side effect https://github.com/scalacenter/scalafix/commit/ad8a4bb481d0a425b01b09a9dadee145c97d3599? That goes far beyong my understanding, but when it was added in https://github.com/scalacenter/scalafix/commit/162db399bbf5584632e617bf7c3135a600dc06c8#diff-0771fc97e5f15a43af8af2a9e65cbc62R137, there was no explicit side effect so I wonder.

Reply:
Correct. This method performs “positioned type checking”, which loads up all the necessary symbols from the classpath. The return value is helpful in other situations like in metals where we use this method to implement parameter hints or completions